File: NemID_CodeGenerator/Code_Generator.py
import requests
import sqlite3

def postRequests():
    db = sqlite3.connect('NemId_CodeGenerator/nem_id_database.sqlite')
    response = requests.post('http://localhost:8090/nemid-auth')
    # Check people in db 
    db_cursor = db.cursor()
    query = """SELECT * FROM user"""

    for row in db_cursor.execute(query):
        if f'NemIdCode: {row[2][-4:]}, NemId: {row[2]}' == response.json():
            json_response={"status":200}
            print(json_response)
        else:
            json_response={"status":403}
            print(json_response)

postRequests()

File: NemID_CodeGenerator/codegenerator.js
const express = require('express');
const bodyParser = require('body-parser')
const sqlite3 = require('sqlite3');

var db = new sqlite3.Database('../NemID_ESB/nem_id_database.sqlite');

let app = express()

app.use(bodyParser.json())

app.post('/nemid-auth', (req, res) => {
    console.log('im here');
    let code = req.body.nemIdCode;
    console.log(code);
    let nemID = req.body.nemId;
    console.log(nemID);

    db.all(`SELECT * FROM user WHERE NemID="${nemID}" AND Password="${code}"`, (err, rows) => {
        if (err || rows.length === 0) {
            return res.status(403).send();
        }

        res.json(
            {
                'generatedCode': Math.floor(Math.random() * 999999 + 100000)
            }
        )
    })

})

app.listen(8090, (err) => {
    if(err){
        console.log(err);
    }
    else{
        console.log("Listening on port 8090");
    }
})